### DISCUSSION TITLE
KPN and Dream Router: IPTV stuck on 85%

### DISCUSSION THREAD

**Discussion by WhizzKid64** [fabianishere/udm-iptv#210] [Q&A] [answered]
Hi there,

First of all, thanks for the great guide and script! This month, I got a KPN fiber connection. It was working great with the KPN Box 12, but I decided to replace that router with the Ubiquiti Dream Router.
However, I don't seem to get IPTV working. Here's what I did:

- I've created a seperate VLAN for IPTV, using VLAN ID 4:
![afbeelding](https://user-images.githubusercontent.com/46891094/207581745-4549dd2e-6ac2-4a4c-af11-ef2666da2fba.png)

- I created the following DHCP option:
![afbeelding](https://user-images.githubusercontent.com/46891094/207581919-dd2c71dd-0742-40c5-a189-bdae78c86572.png)

- The ARRIS VIPS5202 is connected to a "dumb" switch, so I assigned the IPTV Port Profile to Port 2 on the Dream Router itself:

![afbeelding](https://user-images.githubusercontent.com/46891094/207582046-8887e331-be6a-4d97-acf7-6c6a6cc64cec.png)

- I've run Fabian's script through SSH and selected the KPN (NL) configuration profile. I chose VLAN 4 as LAN interface:

![afbeelding](https://user-images.githubusercontent.com/46891094/207582258-9b69cfe5-96df-4dc3-96e2-6acaf3e4084d.png)

I still got this error message:
![afbeelding](https://user-images.githubusercontent.com/46891094/207582354-0aa8121b-22b4-40a6-b707-00d4c4d4911e.png)

I tried fiddling with some settings, enable/disable multicast DNS, used the original KPN DNS servers for the internet connectione etc. Doesn't seem to help.

Here are the diagnostics:

=== Configuration ===
WAN Interface: eth4
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
59: iptv@eth4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.84.198.102/18 brd 10.84.255.255 scope global iptv
       valid_lft forever preferred_lft forever
10.84.192.0/18 proto kernel scope link src 10.84.198.102
213.75.112.0/21 via 10.84.192.1 metric 259
=== Service Logs ===
Dec 14 08:53:08 Dream-Router systemd[1]: Started IPTV support for the UniFi Dream Machine.
Dec 14 08:53:08 Dream-Router udm-iptv[4052]: Obtaining IP address for VLAN interface
Dec 14 08:53:09 Dream-Router udm-iptv[4081]: udhcpc: started, v1.30.1
Dec 14 08:53:09 Dream-Router udm-iptv[4081]: udhcpc: sending discover
Dec 14 08:53:09 Dream-Router udm-iptv[4081]: udhcpc: sending select for 10.84.198.102
Dec 14 08:53:09 Dream-Router udm-iptv[4081]: udhcpc: lease of 10.84.198.102 obtained, lease time 3600
Dec 14 08:53:09 Dream-Router udm-iptv[4052]: NATing IPTV network ranges (if necessary)
Dec 14 08:53:09 Dream-Router udm-iptv[4052]: Setting up igmpproxy
Dec 14 08:53:11 Dream-Router udm-iptv[4052]: Starting igmpproxy
Dec 14 08:53:11 Dream-Router igmpproxy[4052]: No interfaces found for source 169.254.47.193

If somebody has a suggestion for the possible solution, I'm happy to hear it. Thanks in advance!

**Comment by fabianishere** [comment, marked as answer]
The diagnostics are saying you did not select the IPTV traffic to be routed VLAN 4.

When presented the list with internal VLANs to route the IPTV traffic to, you should use spacebar to select items and return to move to the next step.

**Reply by WhizzKid64** [reply]
Yeah, I used 'Enter' after marking the internal VLAN's. Never felt so stupid :-s
I removed udm-iptv and did a reinstall. It's working like a charm! Thank you so much!

Still a question though:
- Should Multicast DNS be disabled or enabled?

**Reply by fabianishere** [reply]
Multicast DNS is not necessary for the IPTV, but should not affect it either.
